src/lsptoolshost/razor/razorEndpoints.ts

```diff
@@ -5,16 +5,76 @@
 
 import { RoslynLanguageServer } from '../server/roslynLanguageServer';
 import * as vscode from 'vscode';
-import { LogMessageParams, NotificationType } from 'vscode-languageclient';
+import {
+    ColorInformation,
+    ColorPresentationParams,
+    ColorPresentationRequest,
+    DocumentColorParams,
+    DocumentColorRequest,
+    LogMessageParams,
+    NotificationType,
+    RequestType,
+} from 'vscode-languageclient';
 import { RazorLogger } from '../../razor/src/razorLogger';
+import { HtmlUpdateParameters } from './htmlUpdateParameters';
+import { UriConverter } from '../utils/uriConverter';
+import { PlatformInformation } from '../../shared/platform';
+import { HtmlDocumentManager } from './htmlDocumentManager';
+import { DocumentColorHandler } from '../../razor/src/documentColor/documentColorHandler';
+import { razorOptions } from '../../shared/options';
+import { ColorPresentationHandler } from '../../razor/src/colorPresentation/colorPresentationHandler';
+import { ColorPresentation } from 'vscode-html-languageservice';
 
 export function registerRazorEndpoints(
     context: vscode.ExtensionContext,
     languageServer: RoslynLanguageServer,
-    razorLogger: RazorLogger
+    razorLogger: RazorLogger,
+    platformInfo: PlatformInformation
 ) {
     const logNotificationType = new NotificationType<LogMessageParams>('razor/log');
     languageServer.registerOnNotificationWithParams(logNotificationType, (params) =>
         razorLogger.log(params.message, params.type)
     );
+
+    if (!razorOptions.cohostingEnabled) {
```

Comment:
Do we need to remove registration of dynamicFile endpoints if cohosting is on?

Reply:
Yeah, I think so. I think we never call them (though tbh I've put almost no effort into initialization yet)

Reply:
can be another pr for sure